src/mpi/coll/transports/gentran/tsp_gentran.c

```diff
@@ -409,7 +409,7 @@ int MPII_Genutil_sched_start(MPII_Genutil_sched_t * sched, MPIR_Comm * comm, MPI
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPII_GENUTIL_SCHED_START);
 
     /* Create a request */
-    reqp = MPIR_Request_create(MPIR_REQUEST_KIND__COLL);
+    reqp = MPIR_Request_create(MPIR_REQUEST_KIND__COLL, 0);
```

Comment:
Will a pool be specified eventually by all components creating requests? For instance, in this example will collective requests become pool aware? I'm trying to wrap my head around the request creation API change.

What I'm getting at is that if request pools will only be used by ch4, then I wonder if `MPIR_Request_create` should stay as it is by hardcoding pool 0, and a separate pool-aware create/free API be added. Of course developers will need to understand when to use the new API. Maybe that's too complex. I'm open to what others think...

Reply:
Pool need be specified by all components. However, component can always choose pool 0 (or any specific pool) and will always be correct -- just mean they will be contending on that pool. Eventually, I think the collectives will also benefit from per-vci locking. For example, user can designate certain collective on specific vci to avoid contention with other threads' pt2pt communications.

Your suggestions could work. However, all requests creation is in a fast path and having explicit pool (vci) argument is good for raising the awareness (of developer). Especially now I created separated paths for request objects from rest of the handlemem allocation, and they may further deviate in the future, so I think let request have distinct interface is not a bad idea.

Reply:
OK, it's good to know that eventually it would make sense for all components to be pool-aware. If others agree, then I think the API change should be OK to merge.